metricbeat/module/elasticsearch/cluster_stats/data_xpack.go

```diff
@@ -0,0 +1,370 @@
+				"open_file_descriptors": c.Dict("open_file_descriptors", s.Schema{
+					"min": c.Int("min"),
+					"max": c.Int("max"),
+					"avg": c.Int("avg"),
+				}),
+			}),
+			"jvm": c.Dict("jvm", s.Schema{
+				"max_uptime_in_millis": c.Int("max_uptime_in_millis"),
+				"mem": c.Dict("mem", s.Schema{
+					"heap_used_in_bytes": c.Int("heap_used_in_bytes"),
+					"heap_max_in_bytes":  c.Int("heap_max_in_bytes"),
+				}),
+				"threads": c.Int("threads"),
+			}),
+			"fs": c.Dict("fs", s.Schema{
+				"total_in_bytes":     c.Int("total_in_bytes"),
+				"free_in_bytes":      c.Int("free_in_bytes"),
+				"available_in_bytes": c.Int("available_in_bytes"),
+			}),
+		}),
+	}
+)
+
+func passthruField(fieldPath string, sourceData, targetData common.MapStr) error {
+	fieldValue, err := sourceData.GetValue(fieldPath)
+	if err != nil {
+		return elastic.MakeErrorForMissingField(fieldPath, elastic.Elasticsearch)
+	}
+
+	targetData.Put(fieldPath, fieldValue)
+	return nil
+}
+
+func clusterNeedsTLSEnabled(license, stackStats common.MapStr) (bool, error) {
+	// TLS does not need to be enabled if license type is something other than trial
+	value, err := license.GetValue("license.type")
+	if err != nil {
+		return false, elastic.MakeErrorForMissingField("license.type", elastic.Elasticsearch)
+	}
+
+	licenseType, ok := value.(string)
+	if !ok {
+		return false, fmt.Errorf("License type is not a string")
+	}
+
+	if licenseType != "trial" {
+		return false, nil
+	}
+
+	// TLS does not need to be enabled if security is not enabled
+	value, err = stackStats.GetValue("security.enabled")
+	if err != nil {
+		return false, elastic.MakeErrorForMissingField("security.enabled", elastic.Elasticsearch)
+	}
+
+	isSecurityEnabled, ok := value.(bool)
+	if !ok {
+		return false, fmt.Errorf("Security enabled flag is not a boolean")
+	}
+
+	if !isSecurityEnabled {
+		return false, nil
+	}
+
+	// TLS does not need to be enabled if TLS is already enabled on the transport protocol
+	value, err = stackStats.GetValue("security.ssl.transport.enabled")
+	if err != nil {
+		return false, elastic.MakeErrorForMissingField("security.ssl.transport.enabled", elastic.Elasticsearch)
+	}
+
+	isTLSAlreadyEnabled, ok := value.(bool)
+	if !ok {
+		return false, fmt.Errorf("Transport protocol SSL enabled flag is not a boolean")
+	}
+
+	return !isTLSAlreadyEnabled, nil
+}
+
+// computeNodesHash computes a simple hash value that can be used to determine if the nodes listing has changed since the last report.
+func computeNodesHash(clusterState common.MapStr) (int32, error) {
+	value, err := clusterState.GetValue("nodes")
+	if err != nil {
+		return 0, elastic.MakeErrorForMissingField("nodes", elastic.Elasticsearch)
+	}
+
+	nodes, ok := value.(map[string]interface{})
+	if !ok {
+		return 0, fmt.Errorf("Nodes is not a map")
+	}
+
+	var nodeEphemeralIDs []string
+	for _, value := range nodes {
+		nodeData, ok := value.(map[string]interface{})
+		if !ok {
+			return 0, fmt.Errorf("Node data is not a map")
```

Comment:
Nit: In general errors which are returned and not printed should start lower case as they could be embeded in other errors. I wonder why hound did not complain. Same applies to most other errors in this PR.